From 0b5b0f4bc6f6c5c8d64cfd04d8d55cefa5509ddc Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Thu, 2 Jul 2020 16:41:22 +0000
Subject: [PATCH] meson: Make Requires in .pc files match autotools

Libreoffice tries to use only harfbuzz-icu.pc and assumes this includes
-lharfbuzz.
---
 src/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/meson.build b/src/meson.build
index 36df8cb13..d3b3246ef 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -501,6 +501,7 @@ pkgmod.generate(libharfbuzz,
 
 pkgmod.generate(libharfbuzz_subset,
   description: 'HarfBuzz font subsetter',
+  requires: [libharfbuzz],
   subdirs: [meson.project_name()],
   version: meson.project_version(),
 )
@@ -531,6 +532,7 @@ if have_icu and not have_icu_builtin
 
   pkgmod.generate(libharfbuzz_icu,
     description: 'HarfBuzz text shaping library ICU integration',
+    requires: [libharfbuzz],
     subdirs: [meson.project_name()],
     version: meson.project_version(),
   )
@@ -644,6 +646,7 @@ if have_gobject
 
   pkgmod.generate(libharfbuzz_gobject,
     description: 'HarfBuzz text shaping library GObject integration',
+    requires: [libharfbuzz, glib_dep, gobject_dep],
     subdirs: [meson.project_name()],
     version: meson.project_version(),
   )
-- 
2.27.0

